code/datums/components/infective.dm

```diff
@@ -8,43 +8,78 @@
 	var/weak_infection_chance = 10
 
 
-/datum/component/infective/Initialize(list/datum/disease/_diseases, expire_in, weak = FALSE)
-	if(islist(_diseases))
-		diseases = _diseases
-	else
-		diseases = list(_diseases)
+/datum/component/infective/Initialize(list/datum/disease/diseases, expire_in, weak = FALSE, weak_infection_chance = 10)
+	if(!ismovable(parent))
+		return COMPONENT_INCOMPATIBLE
+
+	if(!islist(diseases))
+		diseases = islist(diseases)
+
+	///Make sure the diseases list is populated with instances of diseases and nothing else.
```

Comment:
Does this go wrong often? This feels like a band-aid fix

Reply:
Well, I can remove it, I suppose, I was just being paranoid.

Reply:
Ok, no, turns out these lines are necessary to convert typepaths into actual types, so we don't have to task every addcomponent call for this comp to do this.